### ISSUE TITLE
Prebuilding for multiple services using the same image

### ISSUE THREAD

**Issue by ritchiey** [buildkite-plugins/docker-compose-buildkite-plugin#80]
I have multiple services in my docker-compose.yml with the same `image:` property. Only one of them has a `build:` property:

        services:
          app:
            image: my_company/app
            build: .
            ...
          test:
            image: my_company/app

If I just pre-build `app` then when I try to run the `test` service, it rebuilds it in the run step.

If I pre-build `app` and test as described [here](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin#building-multiple-images) then it seems to fail to push `test` to the registry (ECR) and the run step fails because it can't pull the `test` image.

This is the log output I get for the build step:

        $ docker-compose -f docker-compose.yml -p buildkite12345678908243a0af4110f50924cb4e -f docker-compose.buildkite-488-override.yml push app test
        Pushing app (1234567890.dkr.ecr.ap-southeast-2.amazonaws.com/ape-ci/my_companyweb:my_companyweb-app-build-488)...
        The push refers to a repository [1234567890.dkr.ecr.ap-southeast-2.amazonaws.com/ape-ci/my_companyweb]
        0662cafed00d: Pushed
        f9afaae22bee: Pushed
        27d6313d90dc: Layer already exists
        ...
        bee4f8ad5631: Layer already exists
        cafed00d0415: Layer already exists
        deadbeefbcf9: Layer already exists
        my_companyweb-app-build-488: digest: sha256:8c61d757b9c74eca29d9aa74576fafd6ef8581f5e079af3e00f7a24423832138 size: 4098
        $ buildkite-agent meta-data set docker-compose-plugin-built-image-tag-app 1234567890.dkr.ecr.ap-southeast-2.amazonaws.com/ape-ci/my_companyweb:my_companyweb-app-build-488
        $ buildkite-agent meta-data set docker-compose-plugin-built-image-tag-test 1234567890.dkr.ecr.ap-southeast-2.amazonaws.com/ape-ci/my_companyweb:my_companyweb-test-build-488

Which to me looks like it's asking for `test` and `app` to be pushed but it's only pushing `app`.

Any idea how I can avoid having my run step rebuild my image?

Thanks

**Comment by lox** [comment]
Apologies for the slow response, missed this one! I'll try and reproduce this one. Could you confirm what version of the plugin you are on?

**Comment by ritchiey** [comment]
Hi @lox,

I probably should have closed this. I think I understand the correct approach now. I've added both images to the build step even though they're the same image.

Because they're the same image it builds the second one from cache so it happens instantly. It then pushes them both to my CI repository and and then the run steps can pull the image they need.

If there's a better approach, I'd love to hear it but that seems to do the job. I'm on 1.7.0 now but I was on 1.5.2 when I reported this.

Thanks
Ritchie
